Sources/FoundationEssentials/Calendar/Calendar_Cache.swift

```diff
@@ -38,103 +38,71 @@ struct CalendarCache : Sendable {
+    
+    fileprivate init() {
+    }
+    
+    var current: any _CalendarProtocol {
+        if let result = _current.withLock({ $0 }) {
+            return result
         }
+                        
+        let id = Locale.current._calendarIdentifier
+        // If we cannot create the right kind of class, we fail immediately here
+        let calendarClass = CalendarCache.calendarICUClass(identifier: id, useGregorian: true)!
+        let calendar = calendarClass.init(identifier: id, timeZone: nil, locale: Locale.current, firstWeekday: nil, minimumDaysInFirstWeek: nil, gregorianStartDate: nil)
         
-        mutating func autoupdatingCurrent() -> any _CalendarProtocol {
-            if let autoupdatingCurrentCalendar {
-                return autoupdatingCurrentCalendar
+        return _current.withLock {
+            if let current = $0 {
+                // Someone beat us to setting it - use the existing one
+                return current
             } else {
-                let calendar = _CalendarAutoupdating()
-                autoupdatingCurrentCalendar = calendar
+                $0 = calendar
                 return calendar
             }
         }
-
-        mutating func fixed(_ id: Calendar.Identifier) -> any _CalendarProtocol {
-            check()
-            if let cached = fixedCalendars[id] {
-                return cached
-            } else {
-                // If we cannot create the right kind of class, we fail immediately here
-                let calendarClass = CalendarCache.calendarICUClass(identifier: id, useGregorian: true)!
-                let new = calendarClass.init(identifier: id, timeZone: nil, locale: nil, firstWeekday: nil, minimumDaysInFirstWeek: nil, gregorianStartDate: nil)
-                fixedCalendars[id] = new
-                return new
-            }
-        }
-
-        mutating func reset() {
-            wasResetManually = true
-        }
     }
-
-    let lock: LockedState<State>
-
-    static let cache = CalendarCache()
-
-    fileprivate init() {
-        lock = LockedState(initialState: State())
-    }
-
+    
     func reset() {
-        lock.withLock { $0.reset() }
-    }
-
-    var current: any _CalendarProtocol {
-        lock.withLock { $0.current() }
+        // rdar://102017659
+        // Don't create `currentCalendar` here to avoid deadlocking when retrieving a fixed
+        // calendar. Creating the current calendar gets the current locale, decodes a plist
+        // from CFPreferences, and may call +[NSDate initialize] on a separate thread. This
+        // leads to a deadlock if we are also initializing a class on the current thread
+        _current.withLock { $0 = nil }
+        _fixed.withLock { $0 = [:] }
     }
     
-    var autoupdatingCurrent: any _CalendarProtocol {
-        lock.withLock { $0.autoupdatingCurrent() }
-    }
+    // MARK: Singletons
+    
+    static let autoupdatingCurrent = _CalendarAutoupdating()
```

Comment:
For example - this does not need to be stored in the cache, which requires accessing the lock in order to get it.